File: src/main/java/ac/kr/deu/connect/luck/user/UserRole.java
package ac.kr.deu.connect.luck.user;

import org.springframework.security.core.GrantedAuthority;

public enum UserRole implements GrantedAuthority {
    EVENT_MANAGER, USER, ADMIN, FOOD_TRUCK_MANAGER;

    @Override
    public String getAuthority() {
        return "ROLE_" + this.name();
    }
}

File: src/main/kotlin/ac/kr/deu/connect/luck/user/entity/UserRole.kt
package ac.kr.deu.connect.luck.user.entity

import org.springframework.security.core.GrantedAuthority

enum class UserRole : GrantedAuthority {
    USER,
    ADMIN,
    EVENT_MANAGER,
    FOOD_TRUCK_MANAGER,
    ;

    override fun getAuthority(): String {
        return "ROLE_$name"
    }
}
